### PR TITLE
Add typings for plugin-error

### PR DESCRIPTION
**Pull request by rogierschouten**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not provide its own types, and you can not add them.
- [x] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] `tslint.json` should be present, and `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

**Comment by rogierschouten** [comment]
OK I'm stumped.  My first commit gave no lint errors on my machine but it did on Travis. My machine has a fresh NPM install. Can anybody explain to me how to export a class such that you can import it like this?

```typescript
import * as PluginError from "plugin-error";
const p = new PluginError();
```

I tried 'export ='  which works ffine but dts-lint doesn't like it.   I tried 'export default' which of course doesn't work because it's not the same export.  I also tried exporting modules but then travis complains about exporting a module.  

For now, I disabled a lint rule in tslint.json

**Comment by mDibyo** [comment]
I don't think there's any way to import a class like that.

Instead if you just `export default PluginError`, you can import as
```ts
import PluginError from "plugin-error";
```

**Comment by rogierschouten** [comment]
@mDibyo  that is an option if you write a new module but unfortunately the actual module doesn't support doing that.

File: types/plugin-error/plugin-error-tests.ts [hunk 1]
```diff
@@ -0,0 +1,23 @@
+import * as PluginError from 'plugin-error';
```
Comment (armanio123) [inline types/plugin-error/plugin-error-tests.ts]: Based on the plugin-error documentation, use commonjs module instead:
`import PluginError = require('plugin-error');`

**Comment by rogierschouten** [comment]
@paulvanbrenk thanks for merging my ansi-colors typings. FYI I think that despite the discussion above, this PR can be merged

File: types/plugin-error/tslint.json [hunk 2]
```diff
@@ -0,0 +1,6 @@
+{
+    "extends": "dtslint/dt.json",
+    "rules": {
+        "strict-export-declare-modifiers": false
```
Comment (armanio123) [inline types/plugin-error/tslint.json]: Don't disable any linting rules.

**Review by armanio123 — CHANGES_REQUESTED** [review]

**Comment by rogierschouten** [comment]
@armanio123 see discussion above - how can I avoid triggering this lint rule while not resorting to a default export?

**Comment by armanio123** [comment]
I believe the lint rule is due to the export inside the namespace. Just remove that and it should work fine.

File: types/plugin-error/index.d.ts [hunk 3]
```diff
@@ -0,0 +1,56 @@
+// Type definitions for plugin-error 0.1
+// Project: https://github.com/jonschlinkert/plugin-error
+// Definitions by: Rogier Schouten <https://github.com/rogierschouten>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+export = PluginError;
+
+declare namespace PluginError {
+    export interface PluginErrorOpts {
```
Comment (armanio123) [inline types/plugin-error/index.d.ts]: Unnecessary export. You can remove this ones.

**Review by armanio123 — CHANGES_REQUESTED** [review]

**Comment by rogierschouten** [comment]
Removing the exports yields different lint rules:
```
ERROR: 6:1   strict-export-declare-modifiers  Prefer to explicitly write 'export' for an external module export. See: https://github.com/Microsoft/dtslint/blob/master/docs/strict-export-declare-modifiers.md
ERROR: 8:1   strict-export-declare-modifiers  Prefer 'export' to 'declare' in an external module. See: https://github.com/Microsoft/dtslint/blob/master/docs/strict-export-declare-modifiers.md
ERROR: 35:1  strict-export-declare-modifiers  Prefer 'export' to 'declare' in an external module. See: https://github.com/Microsoft/dtslint/blob/master/docs/strict-export-declare-modifiers.md

```

**Comment by armanio123** [comment]
There seems to be an error on dtslint for this. Looks like it's already reported here: https://github.com/Microsoft/dtslint/issues/88

It's ok then to disable the rule while this gets fixed, but I would still suggest to remove the unneeded exports and also import using require.

File: types/plugin-error/tslint.json (hunk 2 above)
Comment (armanio123) [reply types/plugin-error/tslint.json]: Disable is fine as there's an issue with this rule: https://github.com/Microsoft/dtslint/issues/88.

File: types/plugin-error/plugin-error-tests.ts (hunk 1 above)
Comment (rogierschouten) [reply types/plugin-error/plugin-error-tests.ts]: done

**Comment by rogierschouten** [comment]
done

**Pull request merged**